src/main/java/com/mraof/minestuck/block/LootBlock.java

```diff
@@ -0,0 +1,147 @@
+import net.minecraft.block.BlockState;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.item.ItemEntity;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.item.BlockItemUseContext;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.Items;
+import net.minecraft.state.DirectionProperty;
+import net.minecraft.state.StateContainer;
+import net.minecraft.state.properties.BlockStateProperties;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.*;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.BlockRayTraceResult;
+import net.minecraft.util.math.shapes.ISelectionContext;
+import net.minecraft.util.math.shapes.VoxelShape;
+import net.minecraft.util.text.StringTextComponent;
+import net.minecraft.world.Explosion;
+import net.minecraft.world.IBlockReader;
+import net.minecraft.world.IWorldReader;
+import net.minecraft.world.World;
+import net.minecraft.world.storage.loot.LootTables;
+
+import javax.annotation.Nullable;
+import java.util.Map;
+
+public class LootBlock extends Block
+{
+    public static final DirectionProperty FACING = BlockStateProperties.HORIZONTAL_FACING;
+    public final Map<Direction, VoxelShape> shape;
+
+    public LootBlock(Properties properties, CustomVoxelShape shape)
+    {
+        super(properties);
+        this.shape = shape.createRotatedShapes();
+    }
+
+    @Override
+    @SuppressWarnings("deprecation")
+    public boolean onBlockActivated(BlockState state, World worldIn, BlockPos pos, PlayerEntity player, Hand handIn, BlockRayTraceResult hit) {
+        ItemStack itemstack = player.getHeldItem(handIn);
+        if (worldIn.isRemote) {
+            return true;
+        } else {
+            if (itemstack.getItem() == Items.AIR){
+                Direction direction = hit.getFace();
+                Direction direction1 = direction.getAxis() == Direction.Axis.Y ? player.getHorizontalFacing().getOpposite() : direction;
+                worldIn.playSound((PlayerEntity)null, pos, SoundEvents.UI_TOAST_CHALLENGE_COMPLETE, SoundCategory.BLOCKS, 1.0F, 1.0F);
+                if (getBlock() == MSBlocks.CALEDFWLCH_PEDESTAL_FULL){
+                    worldIn.setBlockState(pos, MSBlocks.CALEDFWLCH_PEDESTAL_EMPTY.getDefaultState(), 11);
+                    ItemEntity itementity = new ItemEntity(worldIn, (double)pos.getX() + 0.5D + (double)direction1.getXOffset() * 0.65D, (double)pos.getY() + 0.1D, (double)pos.getZ() + 0.5D + (double)direction1.getZOffset() * 0.65D, new ItemStack(MSItems.CALEDFWLCH, 1));
+                    itementity.setMotion(0.05D * (double)direction1.getXOffset() + worldIn.rand.nextDouble() * 0.02D, 0.05D, 0.05D * (double)direction1.getZOffset() + worldIn.rand.nextDouble() * 0.02D);
+                    worldIn.addEntity(itementity);
+
+                    player.sendMessage(new StringTextComponent("You have claimed the legendary sword Caledfwlch! ...more like a legendary piece of shit."));
```

Comment:
For player messages such as this, use a TranslationTextComponent together with a translation key